### ISSUE TITLE
Unable to start

### ISSUE THREAD

**Issue by maximko** [Tronic/fastrandom#2]
Hello! I found fastrandom on superuser and I can't get it to work. I use Ubuntu 20.04 with latest updates and python version 3.8.10
```
# fastrandom > /dev/null
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
Exception in thread Thread-2:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
Exception in thread Thread-3:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.8/dist-packages/fastrandom/__main__.py", line 19, in gencrypto
    self.run()
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
Exception in thread Thread-4:
  File "/usr/lib/python3.8/threading.py", line 870, in run
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    cipher = Cipher(ChaCha20(key, iv), None) if chacha else Cipher(AES(key), modes.CTR(iv))
Exception in thread Thread-6:
TypeError: __init__() missing 1 required positional argument: 'backend'
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.8/dist-packages/fastrandom/__main__.py", line 19, in gencrypto
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.8/dist-packages/fastrandom/__main__.py", line 19, in gencrypto
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
    cipher = Cipher(ChaCha20(key, iv), None) if chacha else Cipher(AES(key), modes.CTR(iv))
Exception in thread Thread-5:
Exception in thread Thread-7:
TypeError: __init__() missing 1 required positional argument: 'backend'
Exception in thread Thread-9:
Exception in thread Thread-8:
Traceback (most recent call last):
Traceback (most recent call last):
    self._target(*self._args, **self._kwargs)
Traceback (most recent call last):
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
Exception in thread Thread-14:
Exception in thread Thread-15:
Exception in thread Thread-12:
Exception in thread Thread-16:
Exception in thread Thread-10:
Exception in thread Thread-13:
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/fastrandom/__main__.py", line 19, in gencrypto
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
Exception in thread Thread-11:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
Traceback (most recent call last):
Traceback (most recent call last):
    self.run()
    cipher = Cipher(ChaCha20(key, iv), None) if chacha else Cipher(AES(key), modes.CTR(iv))
    cipher = Cipher(ChaCha20(key, iv), None) if chacha else Cipher(AES(key), modes.CTR(iv))
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
TypeError: __init__() missing 1 required positional argument: 'backend'
TypeError: __init__() missing 1 required positional argument: 'backend'
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
Traceback (most recent call last):
    self.run()
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
Traceback (most recent call last):
    self._target(*self._args, **self._kwargs)
    self.run()
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self.run()
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self.run()
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.8/dist-packages/fastrandom/__main__.py", line 19, in gencrypto
  File "/usr/local/lib/python3.8/dist-packages/fastrandom/__main__.py", line 19, in gencrypto
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.8/dist-packages/fastrandom/__main__.py", line 19, in gencrypto
    self.run()
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
    cipher = Cipher(ChaCha20(key, iv), None) if chacha else Cipher(AES(key), modes.CTR(iv))
    self.run()
    self._target(*self._args, **self._kwargs)
    cipher = Cipher(ChaCha20(key, iv), None) if chacha else Cipher(AES(key), modes.CTR(iv))
    cipher = Cipher(ChaCha20(key, iv), None) if chacha else Cipher(AES(key), modes.CTR(iv))
  File "/usr/local/lib/python3.8/dist-packages/fastrandom/__main__.py", line 19, in gencrypto
TypeError: __init__() missing 1 required positional argument: 'backend'
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.8/threading.py", line 870, in run
  File "/usr/local/lib/python3.8/dist-packages/fastrandom/__main__.py", line 19, in gencrypto
    self.run()
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.8/dist-packages/fastrandom/__main__.py", line 19, in gencrypto
TypeError: __init__() missing 1 required positional argument: 'backend'
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.8/dist-packages/fastrandom/__main__.py", line 19, in gencrypto
    cipher = Cipher(ChaCha20(key, iv), None) if chacha else Cipher(AES(key), modes.CTR(iv))
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.8/threading.py", line 870, in run
  File "/usr/lib/python3.8/threading.py", line 870, in run
    cipher = Cipher(ChaCha20(key, iv), None) if chacha else Cipher(AES(key), modes.CTR(iv))
  File "/usr/local/lib/python3.8/dist-packages/fastrandom/__main__.py", line 19, in gencrypto
TypeError: __init__() missing 1 required positional argument: 'backend'
    cipher = Cipher(ChaCha20(key, iv), None) if chacha else Cipher(AES(key), modes.CTR(iv))
TypeError: __init__() missing 1 required positional argument: 'backend'
  File "/usr/local/lib/python3.8/dist-packages/fastrandom/__main__.py", line 19, in gencrypto
    cipher = Cipher(ChaCha20(key, iv), None) if chacha else Cipher(AES(key), modes.CTR(iv))
    self._target(*self._args, **self._kwargs)
TypeError: __init__() missing 1 required positional argument: 'backend'
    self._target(*self._args, **self._kwargs)
    cipher = Cipher(ChaCha20(key, iv), None) if chacha else Cipher(AES(key), modes.CTR(iv))
TypeError: __init__() missing 1 required positional argument: 'backend'
  File "/usr/local/lib/python3.8/dist-packages/fastrandom/__main__.py", line 19, in gencrypto
TypeError: __init__() missing 1 required positional argument: 'backend'
  File "/usr/local/lib/python3.8/dist-packages/fastrandom/__main__.py", line 19, in gencrypto
TypeError: __init__() missing 1 required positional argument: 'backend'
    cipher = Cipher(ChaCha20(key, iv), None) if chacha else Cipher(AES(key), modes.CTR(iv))
  File "/usr/local/lib/python3.8/dist-packages/fastrandom/__main__.py", line 19, in gencrypto
    cipher = Cipher(ChaCha20(key, iv), None) if chacha else Cipher(AES(key), modes.CTR(iv))
TypeError: __init__() missing 1 required positional argument: 'backend'
TypeError: __init__() missing 1 required positional argument: 'backend'
    cipher = Cipher(ChaCha20(key, iv), None) if chacha else Cipher(AES(key), modes.CTR(iv))
    cipher = Cipher(ChaCha20(key, iv), None) if chacha else Cipher(AES(key), modes.CTR(iv))
TypeError: __init__() missing 1 required positional argument: 'backend'
TypeError: __init__() missing 1 required positional argument: 'backend'
[fastrandom]          0 MiB generated at 0.00 GB/s
```

**Comment by Tronic** [comment]
Looks like a compatibility problem with your version of the cryptography module. I've updated the Git version, could you verify if it works with this change?

```
pip install git+https://github.com/Tronic/fastrandom.git
```

**Comment by maximko** [comment]
Thanks, now it generates data but still throwing errors:
```
Exception in thread Thread-2:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
Exception in thread Thread-4:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.8/dist-packages/fastrandom/__main__.py", line 20, in gencrypto
Exception in thread Thread-6:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
Exception in thread Thread-8:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    cipher = Cipher(ChaCha20(key, iv), None) if chacha else Cipher(AES(key), modes.CTR(iv), backend=default_backend())
TypeError: __init__() missing 1 required positional argument: 'backend'
Exception in thread Thread-10:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.8/dist-packages/fastrandom/__main__.py", line 20, in gencrypto
    cipher = Cipher(ChaCha20(key, iv), None) if chacha else Cipher(AES(key), modes.CTR(iv), backend=default_backend())
TypeError: __init__() missing 1 required positional argument: 'backend'
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.8/dist-packages/fastrandom/__main__.py", line 20, in gencrypto
Exception in thread Thread-12:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
    cipher = Cipher(ChaCha20(key, iv), None) if chacha else Cipher(AES(key), modes.CTR(iv), backend=default_backend())
TypeError: __init__() missing 1 required positional argument: 'backend'
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.8/dist-packages/fastrandom/__main__.py", line 20, in gencrypto
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.8/dist-packages/fastrandom/__main__.py", line 20, in gencrypto
    cipher = Cipher(ChaCha20(key, iv), None) if chacha else Cipher(AES(key), modes.CTR(iv), backend=default_backend())
TypeError: __init__() missing 1 required positional argument: 'backend'
Exception in thread Thread-14:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
    cipher = Cipher(ChaCha20(key, iv), None) if chacha else Cipher(AES(key), modes.CTR(iv), backend=default_backend())
TypeError: __init__() missing 1 required positional argument: 'backend'
Exception in thread Thread-16:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.8/dist-packages/fastrandom/__main__.py", line 20, in gencrypto
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.8/dist-packages/fastrandom/__main__.py", line 20, in gencrypto
    cipher = Cipher(ChaCha20(key, iv), None) if chacha else Cipher(AES(key), modes.CTR(iv), backend=default_backend())
TypeError: __init__() missing 1 required positional argument: 'backend'
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.8/dist-packages/fastrandom/__main__.py", line 20, in gencrypto
    cipher = Cipher(ChaCha20(key, iv), None) if chacha else Cipher(AES(key), modes.CTR(iv), backend=default_backend())
TypeError: __init__() missing 1 required positional argument: 'backend'
    cipher = Cipher(ChaCha20(key, iv), None) if chacha else Cipher(AES(key), modes.CTR(iv), backend=default_backend())
TypeError: __init__() missing 1 required positional argument: 'backend'
[fastrandom]     64,463 MiB generated at 2.44 GB/s
```

**Comment by Tronic** [comment]
@maximko Thanks for reporting. I missed another spot where that argument was needed, can you see if it now runs without errors? It was running only half of the workers with the earlier fix, while the other half crashed. Thus, you should be seeing higher performance with the proper fix.

**Comment by maximko** [comment]
Now it runs without errors, thank you!

**Comment by Tronic** [comment]
Fix released in v0.1.0.
